src/constants.js

```diff
@@ -124,3 +124,5 @@ export const VERSION_COLOR_SCHEME: {[key: string]: VersionColor} = {
 };
 
 export type VersionColor = 'red' | 'yellow' | 'green' | 'white';
+
+export type RequestHint = 'dev' | 'optional' | 'resolution' | 'workspaces';
```

Comment:
Don't see other hints being used anywhere else. Do we need anything other than `'dev'` and `'optional'` at the moment?

Reply:
`resolution` is used in: https://github.com/yarnpkg/yarn/blob/5ef113f7db1201f48e22498374023ebe1e61b069/src/cli/commands/install.js#L281

and `workspaces` in: https://github.com/yarnpkg/yarn/blob/5ef113f7db1201f48e22498374023ebe1e61b069/src/cli/commands/install.js#L379